gilded-rose-llc-frontend/pages/index.tsx

```diff
@@ -1,42 +1,27 @@
 import type { NextPage } from 'next'
-import Head from 'next/head'
+import { GetServerSideProps } from 'next'
+import ItemBlock from 'components/item-block'
+import { Item } from 'core/item'
+import * as api from 'core/client'
 
-interface ItemBlockProps {
-  name: string
-  price: string
+export const getServerSideProps: GetServerSideProps = async () => {
+  const gateway = new api.ApiClient(fetch)
+  const items = await gateway.getAllItems()
+  return { props: { items } }
 }
 
-const ItemBlock = (props: ItemBlockProps) => {
-  return (
-    <li className="bg-[#8690F4] my-2">
-      <h2>{props.name}</h2>
-      <h3>{props.price}</h3>
-      <button>add to cart</button>
-    </li>
-  )
+interface HomeProps {
+  items: Item[]
 }
 
-const Home: NextPage = () => {
-  const items = [
-    {
-      name: 'Item Name',
-      price: '$10000000.00',
-    },
-    {
-      name: 'Another Name',
-      price: '$2000000.00',
-    },
-    {
-      name: 'My Bank Account',
-      price: '$0.01',
-    },
-  ]
+const Home: NextPage<HomeProps> = ({ items }) => {
   return (
     <div className="bg-[#222C40] min-h-screen text-white">
-      <h1>GILDED ROSE RULEZ</h1>
+      <h1>Gilded Rose</h1>
       <ul>
-        {items.map((item) => (
-          <ItemBlock key={item.name} name={item.name} price={item.price} />
+        {/* TODO: Will replace the index below with an id located in the items array */}
+        {items.map((item, index) => (
+          <ItemBlock key={index} name={item.name} price={item.price} />
```

Comment:
**Blocker**: We shouldn't use the index as the key when we can use the item's id instead because the order of items is not guaranteed which has performance implications during re-rendering. I suggest reading more about [keys in react](https://reactjs.org/docs/lists-and-keys.html#keys) if you want to learn more.

Reply:
Ah, I saw the comment about using id as the key instead. How much effort do you think it'll be to make that change in this PR?

Reply:
The fastest fix would be to hard code an id into the JSON formatted when it is generated. I would rather not do it this way and think of a more solid implementation while I am working on #30 since that is focused on the Backend.

If you are alright with that. I created issue [#33](https://github.com/jramnani/GildedRoseJavaLLC/issues/33) to handle this

Reply:
Sounds good. Another alternative is to use the item name since we're not expecting two items with the same name right now although we technically could. That's a separate problem though. We probably would want to prioritize add ids to the response since we'll also need it to show individual items.